src/compile/data/formatparse.ts

```diff
@@ -74,6 +80,10 @@ export class ParseNode extends DataFlowNode {
             return;
           }
           parse[fieldDef.field] = 'number';
+        } else if (accessPathDepth(fieldDef.field) > 1) {
+          // For non-date/non-number (strings and booleans), derive a flattened field for a referenced nested field. 
+          // (Parsing numbers / dates already flatten numeric and temporal fields.)
+          parse[fieldDef.field] = 'flatten';
```

Comment:
@domoritz   Don't you have to add the same logic above for `transform`?

Reply:
Side issue: I also wonder if it is behaving correctly for standalone timeUnit transform. 
(the field in `timeUnit` won't get included in parse -- see `line_timeunit_transform` for example.)

Reply:
> Don't you have to add the same logic above for transform -- for all of aggregate, timeUnit, window, filter.

Hmm, yes. I'm surprised we don't do that already. That seems like an orthogonal bug, doesn't it?

Reply:
> Side issue: I also wonder if the original logic above is behaving correctly for standalone timeUnit transform.

It looks correct although we parse the generates date as date again (which is not necessary but we need to track fields to remove it). 

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
  "data": {"url": "data/seattle-weather.csv"},
  "mark": "point",
  "transform": [{
    "timeUnit": "month",
    "field": "foo.date",
    "as": "month"
  }],
  "encoding": {
    "x": {
      "field": "month",
      "type": "temporal"
    }
  }
}
```

```json
{
      "name": "source_0",
      "url": "data/seattle-weather.csv",
      "format": {"type": "csv"},
      "transform": [
        {
          "type": "formula",
          "as": "month",
          "expr": "datetime(0, month(datum[\"foo\"] && datum[\"foo\"][\"date\"]), 1, 0, 0, 0, 0)"
        },
        {
          "type": "formula",
          "expr": "toDate(datum[\"month\"])",
          "as": "month"
        },
        {
          "type": "filter",
          "expr": "datum[\"month\"] !== null && !isNaN(datum[\"month\"])"
        }
      ]
    }
```